src/NuGet.Services.Validation.Orchestrator/ContentScan/ContentScanValidator.cs

```diff
@@ -0,0 +1,142 @@
+        private readonly ContentScanConfiguration _configuration;
+        private readonly ILogger<ContentScanValidator> _logger;
+
+        public ContentScanValidator(
+            IValidationEntitiesContext validationContext,
+            IValidatorStateService validatorStateService,
+            IContentScanEnqueuer contentScanEnqueuer,
+            IOptionsSnapshot<ContentScanConfiguration> configurationAccessor,
+            ILogger<ContentScanValidator> logger)
+        {
+            _validationContext = validationContext ?? throw new ArgumentNullException(nameof(validationContext));
+            _validatorStateService = validatorStateService ?? throw new ArgumentNullException(nameof(validatorStateService));
+            _contentScanEnqueuer = contentScanEnqueuer ?? throw new ArgumentNullException(nameof(contentScanEnqueuer));
+
+            if (configurationAccessor == null)
+            {
+                throw new ArgumentNullException(nameof(configurationAccessor));
+            }
+            if (configurationAccessor.Value == null)
+            {
+                throw new ArgumentException($"{nameof(configurationAccessor.Value)} property is null", nameof(configurationAccessor));
+            }
+            _configuration = configurationAccessor.Value;
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            configurationAccessor = configurationAccessor ?? throw new ArgumentNullException(nameof(configurationAccessor));
+
+            if (configurationAccessor.Value == null)
+            {
+                throw new ArgumentException($"{nameof(configurationAccessor.Value)} property is null", nameof(configurationAccessor));
+            }
+
+            _configuration = configurationAccessor.Value;
+        }
+
+        public async Task<IValidationResponse> StartAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var validatorStatus = await GetValidatorStatusAsync(request);
+
+            if (validatorStatus.State != ValidationStatus.NotStarted)
+            {
+                _logger.LogWarning(
+                    "Scan and Sign validation with validation Id {ValidationStepId} has already started.",
+                    request.ValidationStepId);
+
+                return GetValidationResponse(validatorStatus);
+            }
+
+            if (ShouldSkipScan(request))
+            {
+                return ValidationResponse.Succeeded;
+            }
+
+            await _contentScanEnqueuer.EnqueueContentScanAsync(request.ValidationStepId, request.InputUrl.OriginalString);
+
+            var result = await _validatorStateService.TryAddValidatorStatusAsync(request, validatorStatus, ValidationStatus.Incomplete);
+
+            var response = GetValidationResponse(result);
+            return response;
```

Comment:
I see you used this pattern a few times, but feel free to just return values directly from methods to save a line. For example:

```suggestion
            return GetValidationResponse(result);
```

I don't feel strongly about this, so feel free to keep it as-is if you prefer it :)